cas-management-webapp-support/src/main/java/org/jasig/cas/services/web/factory/DefaultRegisteredServiceFactory.java

```diff
@@ -0,0 +1,176 @@
+package org.jasig.cas.services.web.factory;
+
+import org.jasig.cas.services.AbstractRegisteredService;
+import org.jasig.cas.services.RegisteredService;
+import org.jasig.cas.services.RegisteredServiceAccessStrategy;
+import org.jasig.cas.services.RegisteredServiceAttributeReleasePolicy;
+import org.jasig.cas.services.RegisteredServiceProxyPolicy;
+import org.jasig.cas.services.RegisteredServiceUsernameAttributeProvider;
+import org.jasig.cas.services.web.beans.RegisteredServiceEditBean.FormData;
+import org.jasig.cas.services.web.beans.RegisteredServiceEditBean.ServiceData;
+import org.jasig.cas.services.web.beans.RegisteredServiceViewBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+
+/**
+ * Default implmentation of {@link RegisteredServiceFactory}.
+ *
+ * @author Daniel Frett
+ * @since 4.2
+ */
+@Component
+public final class DefaultRegisteredServiceFactory implements RegisteredServiceFactory {
+    @NotNull
+    @Autowired(required = false)
+    private AccessStrategyMapper accessStrategyMapper = new DefaultAccessStrategyMapper();
+
+    @NotNull
+    @Autowired(required = false)
+    private AttributeFilterMapper attributeFilterMapper = new DefaultAttributeFilterMapper();
+
+    @NotNull
+    @Autowired(required = false)
+    private AttributeReleasePolicyMapper attributeReleasePolicyMapper;
+
+    @NotNull
+    @Autowired(required = false)
+    private PrincipalAttributesRepositoryMapper principalAttributesRepositoryMapper = new
+            DefaultPrincipalAttributesRepositoryMapper();
+
+    @NotNull
+    @Autowired(required = false)
+    private ProxyPolicyMapper proxyPolicyMapper = new DefaultProxyPolicyMapper();
+
+    @NotNull
+    @Autowired(required = false)
+    private RegisteredServiceMapper registeredServiceMapper = new DefaultRegisteredServiceMapper();
+
+    @NotNull
+    @Autowired(required = false)
+    private UsernameAttributeProviderMapper usernameAttributeProviderMapper = new
+            DefaultUsernameAttributeProviderMapper();
```

Comment:
This will help with extensions, because if I write my own impl of PrincipalAttributesRepositoryMapper, Spring will complain that it has found two instances of that, and will fail. Using qualifiers allows me to alias the impl correctly, or use the default beans in some other context. (This is probably why you didnt have Component annotations on the default impls, to avoid this issue)